src/sdk/pynni/nni/compression/torch/utils/mask_conflict.py

```diff
@@ -1,16 +1,187 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
+import os
 import logging
 import torch
 import numpy as np
-from .shape_dependency import ChannelDependency
+from .shape_dependency import ChannelDependency, GroupDependency, CatPaddingDependency
 # logging.basicConfig(level = logging.DEBUG)
 _logger = logging.getLogger('FixMaskConflict')
 
-class MaskConflict:
-    def __init__(self, mask_file, model=None, dummy_input=None, graph=None):
+def fix_mask_conflict(masks, model=None, dummy_input=None, traced=None):
+    """
+    MaskConflict fix the mask conflict for the channel dependencies
+    and group dependency.
+
+    Parameters
+    ----------
+    model : torch.nn.Module
+        model to fix the mask conflict
+    dummy_input : torch.Tensor
+        input example to trace the model
+    masks : dict/str
+        a dict object that stores the masks or the path of the mask file
+    traced : torch._C.torch.jit.TopLevelTracedModule
+        the traced model of the target model, is this parameter is not None,
+        we donnot use the model and dummpy_input to get the trace graph.
+    """
+    if isinstance(masks, str):
+        # if the input is the path of the mask_file
+        assert os.path.exists(masks)
+        masks = torch.load(masks)
+    # if the user uses the model and dummy_input to trace the model, we
+    # should get the traced model handly, so that, we only trace the
+    # model once, GroupMaskConflict and ChannelMaskConflict will reuse
+    # this traced model.
+    if traced is None:
+        assert model is not None and dummy_input is not None
+        with torch.onnx.set_training(model, False):
+            # We need to trace the model in this way, else it will have problems
+            traced = torch.jit.trace(model, dummy_input)
+
+    fix_group_mask = GroupMaskConflict(masks, model, dummy_input, traced)
+    masks = fix_group_mask.fix_mask()
+    fix_channel_mask = ChannelMaskConflict(masks, model, dummy_input, traced)
+    masks = fix_channel_mask.fix_mask()
+    padding_cat_mask = CatMaskPadding(masks, model, dummy_input, traced)
+    masks = padding_cat_mask.fix_mask()
+    return masks
+
+class MaskFix:
+    def __init__(self, masks, model=None, dummy_input=None, traced=None):
+        # check if the parameters are valid
+        parameter_valid = False
+        if traced is not None:
+            parameter_valid = True
+        elif (model is not None) and (dummy_input is not None):
+            parameter_valid = True
+        if not parameter_valid:
+            raise Exception('The input parameters is invalid!')
+        self.model = model
+        self.dummy_input = dummy_input
+        self.traced = traced
+        self.masks = masks
+
+    def fix_mask(self):
+        raise NotImplementedError
+
+    def export(self, path):
+        """
+        Export the masks after fixing the conflict to file.
+        """
+        torch.save(self.masks, path)
+
+class CatMaskPadding(MaskFix):
```

Comment:
could you explain the logic of cat mask padding in docstring to deliver the high level idea of how the conflict is resolved?